feat(aio): copy code snackbar and design updates

sjtrimble opened the pull request:
- Add snackbar and pointer cursor for copy code button inside aio-code components
- Flex cenetered content in features page
- Removed duplicate global css class
- Add styles to links inside of sub-sections
- Remove focus outline on top nav bar links (https://github.com/angular/angular/issues/16097) 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)



**What is the new behavior?**



**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

gkalpak commented on aio/src/app/embedded/code/code.component.ts:
```diff
@@ -65,9 +66,11 @@ export class CodeComponent implements OnChanges {
     private pretty: PrettyPrinter,
     private copier: CopierService,
     private logger: Logger) {}
+    
```
Travis complains about this line having trailing whitespace. It seems you can remove the line altogether.

gkalpak commented on aio/src/app/embedded/code/code.component.ts:
```diff
@@ -65,9 +66,11 @@ export class CodeComponent implements OnChanges {
   @ViewChild('codeContainer') codeContainer: ElementRef;
 
   constructor(
+    private CopyCodeSnackBar: MdSnackBar,
```
This is the generic `SnackBar` service (nothing `CopyCode` specific). I would just name it `snackBar` or `snackBarService`.

gkalpak commented on aio/src/app/embedded/code/code.component.ts:
```diff
@@ -100,6 +103,10 @@ export class CodeComponent implements OnChanges {
       this.buttonLabel = copiedLabel;
       setTimeout(() => this.buttonLabel = originalLabel, 1000);
+      // snackbar alert
+      this.CopyCodeSnackBar.open("Code Copied", "OK", {
```
1. Use single-quotes instead of double-quotes.
2. I don't think we need an action here. It might give the impression that it won't go away unless you click it. And with a 1sec lifespan, it will probably disappear before people read the message and reach out to manually dismiss it.

gkalpak commented on aio/src/app/embedded/code/code.component.ts:
```diff
@@ -100,6 +103,10 @@ export class CodeComponent implements OnChanges {
       setTimeout(() => this.buttonLabel = originalLabel, 1000);
+      // snackbar alert
+      this.CopyCodeSnackBar.open("Code Copied", "OK", {
+      duration: 1000,
```
Wrong indentation 😱

gkalpak commented on aio/src/app/embedded/code/code.component.ts:
```diff
@@ -100,6 +103,10 @@ export class CodeComponent implements OnChanges {
+      duration: 1000,
+      });
     } else {
       this.logger.error('ERROR copying code to clipboard:', code);
```
It might be worth showing a notification if copying failed too.

gkalpak reviewed (changes requested):
We have no tests for the "copy-to-clipboard feature" (but itis not this PR's concern).

Looks good in general, but I would like to also see the preview :grin:

sjtrimble commented:
@gkalpak I cleaned this up and added a separate issue https://github.com/angular/angular/issues/16280 for the test

petebacondarwin commented on aio/src/app/embedded/code/code.component.ts:
```diff
@@ -100,8 +102,16 @@ export class CodeComponent implements OnChanges {
       // change the button label (for one second)
       this.buttonLabel = copiedLabel;
       setTimeout(() => this.buttonLabel = originalLabel, 1000);
```
we need to remove this unused code now (in another PR)

sjtrimble committed:
feat(aio): copy code snackbar and design updates

- Add snackbar and pointer cursor for copy code button inside aio-code components
- Flex cenetered content in features page
- Removed duplicate global css class
- Add styles to links inside of sub-sections
- Remove focus outline on top nav bar links

petebacondarwin commented:
I have fixed and added tests.

gkalpak commented on aio/src/app/embedded/code/code.component.ts:
```diff
@@ -97,11 +94,16 @@ export class CodeComponent implements OnChanges {
     } else {
       this.logger.error('ERROR copying code to clipboard:', code);
+      // failure snackbar alert
+      this.snackbar.open('Copy failed. Please try again!', 'OK', {
```
Nit: Unnecessary action?

petebacondarwin replied:
Not sure?

petebacondarwin replied:
Too late you approved it now :-)

gkalpak replied:
Not sure if you thought about it and decided to add an action anyway (in which case I am fine with it) or you haven't.

sjtrimble replied:
I just updated .. sorry missed that one when I was testing

petebacondarwin replied:
Doh! That broke the unit tests again @sjtrimble :-)

mary-poppins commented:
The angular.io preview for 631feab365b7b89b541b2f4f2e25b5a4286d4191 is available [here][1].

[1]: https://pr16268-631feab365b7b89b541b2f4f2e25b5a4286d4191.ngbuilds.io/

gkalpak reviewed (approved):
LGTM

petebacondarwin commented:
I have fixed the test and squashed.

mary-poppins commented:
The angular.io preview for 75bc11bfdc97c3f33532135fcda80da8cdae6d5f is available [here][1].

[1]: https://pr16268-75bc11bfdc97c3f33532135fcda80da8cdae6d5f.ngbuilds.io/
